backend/de.metas.contracts/src/main/java/de/metas/contracts/bpartner/command/ContractLocationReplaceCommand.java

```diff
@@ -70,10 +73,22 @@ private void updateFlatrateTermColumn(final String columnName, final RepoIdAware
 	{
 		final ICompositeQueryUpdater<I_C_Flatrate_Term> queryUpdater = queryBL.createCompositeQueryUpdater(I_C_Flatrate_Term.class)
 				.addSetColumnValue(columnName, newLocationId);
-		
+
+		final IQueryFilter<I_C_Flatrate_Term> filterEndDateGreaterThanToday = queryBL.createCompositeQueryFilter(I_C_Flatrate_Term.class)
+				.addCompareFilter(I_C_Flatrate_Term.COLUMNNAME_EndDate, CompareQueryFilter.Operator.GREATER, Env.getDate());
+
+		final IQueryFilter<I_C_Flatrate_Term> filterEndDateNull = queryBL.createCompositeQueryFilter(I_C_Flatrate_Term.class)
+				.addEqualsFilter(I_C_Flatrate_Term.COLUMNNAME_EndDate, null);
+
+		final IQueryFilter<I_C_Flatrate_Term> contractFilters = queryBL.createCompositeQueryFilter(I_C_Flatrate_Term.class)
+				.setJoinOr()
+				.addFilter(filterEndDateGreaterThanToday)
+				.addFilter(filterEndDateNull);
+
```

Comment:
```suggestion

		final IQueryFilter<I_C_Flatrate_Term> endDatefilters = queryBL.createCompositeQueryFilter(I_C_Flatrate_Term.class)
				.setJoinOr()
				.addCompareFilter(I_C_Flatrate_Term.COLUMNNAME_EndDate, CompareQueryFilter.Operator.GREATER, Env.getDate())
				.addEqualsFilter(I_C_Flatrate_Term.COLUMNNAME_EndDate, null);
```

there's no reason to separate them like that